datasets/circa/circa.py

```diff
@@ -0,0 +1,154 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Dataset containing polar questions and indirect answers."""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+
+import datasets
+
+
+_CITATION = """\
+@InProceedings{louis_emnlp2020,
+  author =      "Annie Louis and Dan Roth and Filip Radlinski",
+  title =       ""{I}'d rather just go to bed": {U}nderstanding {I}ndirect {A}nswers",
+  booktitle =   "Proceedings of the 2020 Conference on Empirical Methods
+  in Natural Language Processing",
+  year =        "2020",
+}
+"""
+
+_DESCRIPTION = """\
+The Circa (meaning ‘approximately’) dataset aims to help machine learning systems
+to solve the problem of interpreting indirect answers to polar questions.
+
+The dataset contains pairs of yes/no questions and indirect answers, together with
+annotations for the interpretation of the answer. The data is collected in 10
+different social conversational situations (eg. food preferences of a friend).
+
+NOTE: There might be missing labels in the dataset and we have replaced them with -1.
+The original dataset contains no train/dev/test splits.
+"""
+
+_LICENSE = "Creative Commons Attribution 4.0 License"
+
+_DATA_URL = "https://raw.githubusercontent.com/google-research-datasets/circa/main/circa-data.tsv"
+
+
+class Circa(datasets.GeneratorBasedBuilder):
+    """Dataset containing polar questions and indirect answers."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "context": datasets.Value("string"),
+                "question-X": datasets.Value("string"),
+                "canquestion-X": datasets.Value("string"),
+                "answer-Y": datasets.Value("string"),
+                "judgements": datasets.Value("string"),
```

Comment:
We could make `judgements` a sequence of labels by splitting on the `#` if I understand correctly. What do you think ?

Reply:
In their paper and in the dataset, ```judgement``` is a combination of results which 5 annotators marked separated by```#```. They use majority judgement as the gold standard, subject to at least three annotators making that judgement. I reckon they'll eventually need gold standard for the question-answer-pair-classification task. Hence, I marked its value as string and not sequence of labels.

Reply:
I will be taking action agony all of this spying that has been requested my a company that does not exists. AKA boyfriend. Which you should be checking into

Reply:
> I will be taking action agony all of this spying that has been requested my a company that does not exists. AKA boyfriend. Which you should be checking into